hanadb_exporter/exporters/prometheus_exporter.py

```diff
@@ -31,48 +32,77 @@ class SapHanaCollector(object):
     SAP HANA database data exporter
     """
 
-    def __init__(self, conector):
+    def __init__(self, connector):
         super(SapHanaCollector, self).__init__()
         self._logger = logging.getLogger(__name__)
-        self._hdb_connector = conector
+        self._hdb_connector = connector
 
-    def _execute(self, metric):
+    def _execute(self, query):
         """
-        Create metric object
+        Execute query and return a QueryResult object
 
         Args:
-            metric (dict): query, info, type structure dictionary
+            query (string): SQL query
         """
         try:
-            value = self._hdb_connector.query(metric['query'])
-
-            if metric['type'] == core.GaugeMetricFamily:
-                metric_obj = self._manage_gauge(metric, value)
-            else:
-                raise NotImplementedError('{} type not implemented'.format(metric['type']))
-
-            return metric_obj
+            query_result = self._hdb_connector.query(query)
+            return query_result
         except KeyError as err:
             raise MalformedMetric(err)
 
-    def _manage_gauge(self, metric, value):
+    def _format_query_result(self, query_result):
+        """
+        Format query results to match column names with their values for each row
+        Returns nested list, containing tuples (column_name, value)
+
+        Args:
+            query_result (obj): QueryResult object
+        """
+        query_columns = []
+        formatted_query_result = []
+        for meta in query_result.metadata:
```

Comment:
We could just us a comprehension list here to create the query_columns. It is more efficient and it looks better.
Something like:

```
query_columns = [meta[0] for meta in query_reulst.metadata]
```